crypto/multisig/src/lib.rs

```diff
@@ -445,6 +504,184 @@ mod test {
+            vec![individual_signer.public_key()],
+            vec![org1_signer_set.clone(), org2_signer_set.clone()],
+            3,
+        );
+
+        let multi_sig = MultiSig::new(vec![
+            // Org 1 satisfied
+            org1_signer1.try_sign(message.as_ref()).unwrap(),
+            org1_signer2.try_sign(message.as_ref()).unwrap(),
+            // Org 2 unsatisfied
+            org2_signer1.try_sign(message.as_ref()).unwrap(),
+            // Individual signer satisfied
+            individual_signer.try_sign(message.as_ref()).unwrap(),
+        ]);
+        assert!(signer_set.verify(message.as_ref(), &multi_sig).is_err());
+
+        let multi_sig = MultiSig::new(vec![
+            // Org 1 satisfied
+            org1_signer1.try_sign(message.as_ref()).unwrap(),
+            org1_signer2.try_sign(message.as_ref()).unwrap(),
+            // Org 2 satisfied
+            org2_signer1.try_sign(message.as_ref()).unwrap(),
+            org2_signer2.try_sign(message.as_ref()).unwrap(),
+        ]);
+        assert!(signer_set.verify(message.as_ref(), &multi_sig).is_err());
+
+        let multi_sig = MultiSig::new(vec![
+            // Org 1 satisfied
+            org1_signer1.try_sign(message.as_ref()).unwrap(),
+            org1_signer2.try_sign(message.as_ref()).unwrap(),
+            // Org 2 satisfied (and has an extra signature, but this still doesn't count towards
+            // the top level threshold)
+            org2_signer1.try_sign(message.as_ref()).unwrap(),
+            org2_signer2.try_sign(message.as_ref()).unwrap(),
+            org2_signer3.try_sign(message.as_ref()).unwrap(),
+        ]);
+        assert!(signer_set.verify(message.as_ref(), &multi_sig).is_err());
+
+        // Providing the same signers multiple times does not change the result.
+        let multi_sig = MultiSig::new(vec![
+            // Org 1 satisfied
+            org1_signer1.try_sign(message.as_ref()).unwrap(),
+            org1_signer2.try_sign(message.as_ref()).unwrap(),
+            org1_signer2.try_sign(message.as_ref()).unwrap(),
+            org1_signer2.try_sign(message.as_ref()).unwrap(),
+            // Org 2 satisfied
+            org2_signer1.try_sign(message.as_ref()).unwrap(),
+            org2_signer2.try_sign(message.as_ref()).unwrap(),
+            org2_signer1.try_sign(message.as_ref()).unwrap(),
+            org2_signer2.try_sign(message.as_ref()).unwrap(),
+        ]);
+        assert!(signer_set.verify(message.as_ref(), &multi_sig).is_err());
+
+        let multi_sig = MultiSig::new(vec![
+            // Org 1 satisfied
+            org1_signer1.try_sign(message.as_ref()).unwrap(),
+            org1_signer2.try_sign(message.as_ref()).unwrap(),
+            // Org 2 satisfied
+            org2_signer1.try_sign(message.as_ref()).unwrap(),
+            org2_signer2.try_sign(message.as_ref()).unwrap(),
+            // Individual signer satisfied
+            individual_signer.try_sign(message.as_ref()).unwrap(),
+        ]);
```

Comment:
Optional/DRY: Extract a test helper to generate a `MultiSig`
```suggestion
        let multi_sig = make_multisig(message.as_ref(), [
            // Org 1 satisfied
            &org1_signer1,
            &org1_signer2,
            // Org 2 satisfied
            &org2_signer1,
            &org2_signer2,
            // Individual signer satisfied
            &individual_signer, 
        ]);
```

with
```rust
fn make_multi_sig<S>(message: &[u8], signers: [&impl Signer<S>]) -> MultiSig<S>
where
    S: Clone + Default + Digestible + Eq + Hash + Message + Ord + Serialize + Signature,
{
    MultiSig::new(
        signers
            .into_iter()
            .map(|s| s.sign(message))
            .collect()
    )
}
```